src/Piece/Queen.cpp

```diff
@@ -0,0 +1,18 @@
+#pragma once
+#include "Piece/Queen.h"
+
+bool Queen::isValidMove(const Position src, const Position des) const {
+    // Check if the move is along a straight line (horizontally, vertically, or diagonally)
+    if (src.row == des.row || src.col == des.col || abs(src.row - des.row) == abs(src.col - des.col)) {
+        // The move is along a straight line
+        return true;
```

Comment:
Simplify the condition: Instead of explicitly returning true or false, you can directly return the result of the condition